accounts-db/src/ancient_append_vecs.rs

```diff
@@ -293,15 +293,51 @@ impl AccountsDb {
                 &ancient_slot_infos.all_infos[..],
             );
 
-        let accounts_to_combine = self.calc_accounts_to_combine(&accounts_per_storage);
+        let mut accounts_to_combine = self.calc_accounts_to_combine(&accounts_per_storage);
```

Comment:
When we are combining many-ref accounts from old slot to the new highest slot, it is possible that the new highest slot is on a fork, and later that fork was abandoned and never rooted. Then `clean` will drop that slot. And we will lose the account from the old slot?

Reply:
No, the slots considered here are all an epoch older than the max root (give or take an offset). So, they are all roots.